identity-core/src/convert/json.rs

```diff
@@ -53,19 +53,57 @@ impl<T> ToJson for T where T: Serialize {}
 /// A convenience-trait for types that can be deserialized from JSON.
 pub trait FromJson: for<'de> Deserialize<'de> + Sized {
   /// Deserialize `Self` from a string of JSON text.
-  fn from_json(json: &(impl AsRef<str> + ?Sized)) -> Result<Self> {
-    serde_json::from_str(json.as_ref()).map_err(Error::DecodeJSON)
+  fn from_json(json: &(impl AsRef<str> + ?Sized)) -> Result<Self, serde_json::Error> {
+    serde_json::from_str(json.as_ref())
   }
 
   /// Deserialize `Self` from bytes of JSON text.
-  fn from_json_slice(json: &(impl AsRef<[u8]> + ?Sized)) -> Result<Self> {
-    serde_json::from_slice(json.as_ref()).map_err(Error::DecodeJSON)
+  fn from_json_slice(json: &(impl AsRef<[u8]> + ?Sized)) -> Result<Self, serde_json::Error> {
+    serde_json::from_slice(json.as_ref())
   }
 
   /// Deserialize `Self` from a [`serde_json::Value`].
-  fn from_json_value(json: serde_json::Value) -> Result<Self> {
-    serde_json::from_value(json).map_err(Error::DecodeJSON)
+  fn from_json_value(json: serde_json::Value) -> Result<Self, serde_json::Error> {
+    serde_json::from_value(json)
   }
 }
 
 impl<T> FromJson for T where T: for<'de> Deserialize<'de> + Sized {}
+
+mod errors {
+  use thiserror::Error as DeriveError;
+  /*
+  We implement From serde_json::Error for the errors in this module because we want the ToJson and FromJson traits to be easy
+  to implement downstream and serde_json is both stable and has millions of downloads hence
+  it is relatively safe with regards to stability to include this error type in our public API.
+  For more interoperability we also implement Into serde_json::Error
```

Comment:
I started using `serde_json::Error`, but I found it useful later on to be able to encode whether the error occurred during serialization or deserialization in the error type itself, which is why I introduced two new wrapper types here.

Reply:
Do you think two type aliases would also work?

Reply:
Nice suggestion. I will look into this.

Reply:
Since it can be convenient to convert these errors and pass on the information that it was a serialization, or deserialization error during type transformation I will keep these two types for now.